### ISSUE TITLE
Gallery: Add inline reordering for images within block

### ISSUE THREAD

**Issue by mkaz** [WordPress/gutenberg#2129]
The gallery block relies on the core media modal dialog for reordering images within the block, this should be managed from within the block itself.

Add ability to drag and drop reorder images within the block

**Comment by afercia** [comment]
> Add ability to drag and drop reorder images within the block

Users should be able to do this also when using only the keyboard, i.e. drag and drop needs an alternative UI.

**Comment by mtias** [comment]
Closing as duplicate of #743.
